### PR TITLE
Fix Http client cancellation

### PR DESCRIPTION
**Pull request by drewhk**
- fixed splitWhen cancellation
- added eagerCancel mode to Broadcast
- made Http use eagerCancel in client
- removed wrong GraphJunctionAtrributesSpec that fails always

### CODE REVIEW

File: akka-http-core/src/test/scala/akka/http/impl/engine/client/ClientCancellationSpec.scala [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    "support cancellation in pooled outgoing connection" in {
+      testCase(
+        Flow[HttpRequest]
+          .map((_, ()))
+          .via(Http().cachedHostConnectionPool(address.getHostName, address.getPort)(noncheckedMaterializer))
+          .map(_._1.get))
+    }
+
+    "support cancellation in simple outgoing connection with TLS" in {
+      pending
```
Comment (drewhk) [inline akka-http-core/src/test/scala/akka/http/impl/engine/client/ClientCancellationSpec.scala]: I cannot figure out how to make the HTTPS server to work, hence pending

**Comment by sirthias** [comment]
Nice. LGTM.
I like the `eagerClose` flag.

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowSplitWhenSpec.scala [hunk 2]
```diff
@@ -114,6 +114,72 @@ class FlowSplitWhenSpec extends AkkaSpec {
...
+      inputs3.expectNoMsg(1.second)
+
+      substream4.cancel()
+      inputs3.expectNoMsg(1.second)
+      masterStream3.expectNoMsg(1.second)
+
+      masterStream3.cancel()
+      inputs3.expectCancellation()
+
+    }
```
Comment (rkuhn) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowSplitWhenSpec.scala]: shouldn’t these be separate test cases?

Comment (drewhk) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowSplitWhenSpec.scala]: They all test the same thing

Comment (drewhk) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowSplitWhenSpec.scala]: I just tried many variations to be sure and wanted to keep all.

**Comment by rkuhn** [comment]
LGTM

**Comment by drewhk** [comment]
wonderful :)

**Pull request merged**